### PR TITLE
remove or fix broken links in documentation

### PR DESCRIPTION
**Pull request by yzhaobom**
Documentation fixes for broken links

- [x] If a new developer, signed up to CLA

### CODE REVIEW

File: doc/source/about.rst [hunk 1]
```diff
@@ -82,7 +81,7 @@ This simple example covers one step in a processing chain. Additional informatio
 
 The current list of CLIs can be `found here`_.
 
-.. _found here: improver.cli.html
+.. _found here: https://improver.readthedocs.io/en/latest/improver.cli.html
```
Comment (tjtg) [inline doc/source/about.rst:85]: Suggest avoiding hardcoded link to readthedocs for internal links within the documentation. This makes the link work with an offline copy of the built documentation.
For example, ```The current list of CLIs can be :doc:`found here <improver.cli>\`.```

The other links above to environment YAMLs need to be external github links as the YAMLs aren't included in the documentation build.

**Review by tjtg — COMMENTED** [review]
Thanks for submitting fixes to these broken links in the documentation. I've suggested a neater way to make the last change, all the rest looks good.

Please also add another commit to add your name to the list of contributors in CONTRIBUTING.md

File: doc/source/about.rst (hunk 1 above)
Comment (yzhaobom) [reply doc/source/about.rst]: @tjtg as discussed, I revert to original about.rst

**Comment by yzhaobom** [comment]
@tjtg, done as required.

**Review by tjtg — APPROVED** [review]
All looks good now.

**Pull request merged**